### PR TITLE
new endpoint POST /api/users/{id}/accounts

### PR DESCRIPTION
**Pull request by Schparky**

### CODE REVIEW

File: application/models/user.go [hunk 1]
```diff
@@ -177,12 +193,20 @@ func (u *User) Login(c buffalo.Context, profile auth.Profile) error {
 	return nil
 }
 
-func (u *User) LoadSubscription(tx *pop.Connection, reload bool) {
-	if u.Subscription.ID.IsNil() || reload {
-		if err := tx.Load(u); err != nil {
-			panic("database error loading User.Subscription, " + err.Error())
-		}
+func (u *User) CreateAccount(tx *pop.Connection, input api.AccountCreateInput) error {
+	if u.RecurlyID.Valid {
```
Comment (Baggerone) [inline application/models/user.go:197]: For some reason, I'm concerned we may need an additional check here. 
Should we be careful and ensure that the u.ID isn't a zero UUID? 
That might help avoid an issue in the future if we ever were to call the Recurly api with an empty user.

**Review by Baggerone — APPROVED** [review]
Looking good. I included one question.

File: application/models/user.go (hunk 1 above)
Comment (Schparky) [reply application/models/user.go:197]: Does it help to know that the AuthZ middleware has already checked it?

Comment (Baggerone) [reply application/models/user.go:197]: It helps in theory, but it's possible for this method to be called from somewhere where Authz isn't in play, e.g. from another model or a test.

If you're not worried, that's OK. We can leave it as is.

**Review by jason-jackson — COMMENTED** [review]
What happens if a user has a recurly account, but for some reason it isn't in the database?

Say for example, recurly api is called, but for the database update had an error.

**Comment by Schparky** [comment]
> What happens if a user has a recurly account, but for some reason it isn't in the database?
> 
> Say for example, recurly api is called, but for the database update had an error.

Good question. Then they would get a second account and someone would have to manually merge the two. Better to try to avoid that situation if possible. I'll keep it in mind when I actually implement the Recurly integration.

File: application/models/user.go (hunk 1 above)
Comment (Schparky) [reply application/models/user.go:197]: Not too worried, but thanks for bringing it up. It's kind of in the same category as checking the receiver for nil. We should, in theory, but rarely do.

**Pull request merged**